src/vs/workbench/contrib/terminal/electron-browser/terminalService.ts

```diff
@@ -117,7 +117,30 @@ export class TerminalService extends BrowserTerminalService implements ITerminal
 		});
 	}
 
-	private _detectWindowsShells(): Promise<IQuickPickItem[]> {
+	/**
+	 * Get the executable file path of shell from registry.
+	 * @param Registry The data of imported from `vscode-windows-registry`
+	 * @param shellName The shell name to get the executable file path
+	 * @returns The executable file path of shell or `'ShellNotFound'`
+	 */
+	private _getShellPathFromRegistry(Registry: typeof import('vscode-windows-registry'), shellName: string): string {
+		const shellNotFound = 'ShellNotFound';
+		let shellPath;
+
+		try {
+			shellPath = Registry.GetStringRegKey('HKEY_LOCAL_MACHINE', `SOFTWARE\\Microsoft\\Windows\\CurrentVersion\\App Paths\\${shellName}.exe`, '');
```

Comment:
This could be simplified:

```
try {
  return (grab from reg);
} catch () {
}
return [];
```

Reply:
`Registry.GetStringRegKey()` function returns `string` or `undefined`.
https://github.com/Microsoft/vscode/blob/3a12b7ac2efd2f1f01a646ef5272313ad50bf618/src/typings/vscode-windows-registry.d.ts#L8
Therefore, this function needs a logic to convert from `undefined` to `[]`.
But I referred your review when I fixed the code.

Thanks.

Reply:
I realized that using `!` makes the logic unnecessary. So I deleted the logic.

sorry for the trouble!